### DISCUSSION TITLE
Querying Most Popular Tweets using Tweepy V2 API

### DISCUSSION THREAD

**Discussion by StephenW789** [tweepy/tweepy#1955] [Questions] [answered]
I am scraping Twitter tweets using the Tweepy V2 API, but am having trouble querying for more popular tweets with higher retweets and likes.

My application requires that I ignore retweets and replies for tweets via `-is:retweet -is:reply`. I am searching for the original tweets themselves.

However, I currently have numerous tweets which have very low likes and retweets, which is inadequate for my application.

Is there any method to query based on popularity or retweets count in Tweepy V2? The popularity filter seems to only exist in Tweepy V1 (https://stackoverflow.com/questions/44447258/is-there-a-way-to-search-for-top-tweets-with-tweepy-instead-of-latest-tweets).

Sample code is shown below:
```
query_str = "#chosen_topic lang:en -is:retweet -is:reply"

client = tweepy.Client(bearer_token=config.BEARER_TOKEN, consumer_key=
config.CONSUMER_KEY,consumer_secret= config.CONSUMER_SECRET,access_token=
config.ACCESS_TOKEN,access_token_secret= config.ACCESS_TOKEN_SECRET)

for tweet_batch in tweepy.Paginator(client.search_all_tweets, query=query_str,
                                    tweet_fields=['context_annotations','created_at', 'public_metrics','author_id', 'lang', 'geo', 'entities'], 
                                    user_fields=['name','username','location','verified','description'],
                                    max_results=100, expansions='author_id'):

    tweet_data = tweet_batch.data
    for tweet in tweet_data:
        print(tweet.public_metrics['reply_count'])
```

**Comment by Harmon758** [comment, marked as answer]
You're referring to Twitter API v2 and Twitter API v1.1, as Tweepy v2 was released almost a decade ago and Tweepy v1 is from over a decade ago.

I don't think there's a good way to do this with Twitter API v2, beyond trying setting the `sort_order` parameter to `"relevancy"`.

You can provide feedback on Twitter API v2 functionality on the [Twitter Developers Forums](https://twittercommunity.com/).
